Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/run.go

```diff
@@ -148,6 +156,14 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
 	}
 
 	generatorName := cmdutil.GetFlagString(cmd, "generator")
+	if len(generatorName) == 0 && opts != nil {
```

Comment:
I would be careful with this carry, with the changes I've introduced. Because current code allows nice fallback to old behavior if no new resources are available. I'd rather see this being hooked into the discover logic below, imho.